nes-network/network/src/sender/control.rs

```diff
@@ -0,0 +1,453 @@
+) -> Result<()> {
+    let (request_connection, mut await_connection_request) = tokio::sync::mpsc::channel::<
+        oneshot::Sender<(
+            ControlChannelSenderReader<C::Reader>,
+            ControlChannelSenderWriter<C::Writer>,
+        )>,
+    >(1);
+
+    let (channel_registration_request_handler, mut await_channel_registration_request) =
+        tokio::sync::mpsc::channel::<(ChannelIdentifier, oneshot::Sender<EstablishChannelResult>)>(
+            10,
+        );
+
+    // Connection Keepalive Task:
+    // This task is responsible for establishing connections to the target_connection.
+    // It implements exponential backoff.
+    // Once a connection has been established and transfers the Reader/Writer pair to `connection_tx`
+    // `connection_tx` also receives an oneshot channel to signal that the keepalive task should
+    // once again retry to establish the connection.
+    let _connection_keep_alive = ScopedTask::new(tokio::spawn(
+        {
+            let target_connection = target_connection.clone();
+            let this_connection = this_connection.clone();
+            let communication = communication.clone();
+            async move {
+                loop {
+                    let Some(await_connection) = await_connection_request.recv().await else {
+                        return;
+                    };
+
+                    let retry = ExponentialBackoff::from_millis(2)
+                        .max_delay_millis(500)
+                        .map(jitter);
+
+                    let connection = Retry::spawn(retry, || async {
+                        create_connection(
+                            this_connection.clone(),
+                            target_connection.clone(),
+                            communication.clone(),
+                        )
+                            .await
+                    })
+                        .await;
+                    let (reader, writer) = match connection {
+                        Ok(connection) => connection,
+                        Err(e) => {
+                            error!(
+                                "Could not establish connection to {}: {e:?}",
+                                target_connection
+                            );
+                            return;
+                        }
+                    };
+
+                    info!("Connection to {} was established", target_connection);
+                    let _ = await_connection.send((reader, writer));
+                }
+            }
+        }
+            .in_current_span(),
+    ));
+
+    // Handling Request on the NetworkingConnectionListener is broken up into two separate tasks.
+    // This task is only responsible for establishing a new DataChannel once they have been accepted
+    // by the other side.
+    tokio::spawn({
+        async move {
+            'connection: loop {
+                let (tx, rx) = oneshot::channel();
+                request_connection
```

Comment:
```suggestion
                // Ask the control channel to negotiate a new data channel
                request_connection
```

Is this what happens?